### ISSUE TITLE
Download on-demand per folder

### ISSUE THREAD

**Issue by ferdiga** [owncloud/client#2633]
Hello!
It would be helpful to create a "no automatic download" option for the desktop client similar to android client. on a per folder basis

Download only files which are accessed (and store them)

Reason
a user might need access to a lot of data, but does not want to download them all (especially not immediately after install)

**Comment by oliverdo** [comment]
Hmmm,

I'm unsure about it, but I will give it a test. However using the ignore option from the client as a workaround to decide which data will be synchronized will not give me a way/report/overview to see which data is on which client.

The basic idea behind my question is, I want to decide which parts of my data should be stored on which client.

So I see multiple steps.
1st, I have my to partition my data, like photos, music, video, family, work, etc. (this could be done by folders)
2nd, I need a list of devices which have an account that access my data
3rd, I need some type of GUI there I could configure which parts of my data will be on which device accessible and if it should be downloaded automatically
4th, (in a perfect world) I could see how many percent’s of my data in oc are already synchronized to which device

I think this is something that OC could not deliver at the moment. From my point of view this is an advanced feature makes it really my decision where to store my data

What do you think?
Oliver

**Comment by oliverdo** [comment]
Hmm,

I just saw that I could configure the top level folders during the initial configuration of the client, which is quite near to what I want. However this is a client configuration and has two disadvantages.
1.	There is no configuration on the client to change it afterwards
2.	There seems to be no way to get a central look at the client configuration
The reason why I’m asking for that is that the number of devices even inside a family (and of course also at work) is growing.

At this moment we have 4 computers and one smartphone inside the part of family that lives at my house. There also some accounts and devices that belongs to other family members that live not wat my house.

At work I have two computers and one smartphone just for my work.

So I do not need all data on all devices, at the same is correct for all others and therefore I searching for a way to get a central look and administration of it.

Greetings
Oliver

**Comment by ferdiga** [comment]
"I do not need all data STORED on all devices" is the key argument

**Comment by lionax** [comment]
Hey I would like to participate in this discussion.

I have just setup my ownCloud and wanted to use it for document synchronization but also for media access on demand like music. I simply don't want to download my whole music library to my small ssd of my notebook but listen to it when needed.

Therefore it would be nice, if I just could setup my music folder in the desktop client to show my library without downloading the whole thing. Just streaming the media I'm accessing.

Additionally I'm editing all the media with mediamonkey and mp3tag (I know that's a matter of taste) and would like to keep using this programs. So would it be possible to have access to the media files for listening and editing but to download only the necessary files?

Sure there is WebDAV (don't know if it would do what I want yet) and Ampache for streaming but a integrated solution in the client would be really nice ;)

**Comment by danimo** [comment]
You guys are asking for a feature that can only be implemented on a file system level. The idea that people are having of opening a dummy file and downloading it on demand is simply not possible without implementing a custom file system.

What we can do so far is implemented via selective sync. Anything beyond this is out of scope for the forseeable future

**Comment by TomFernandes** [comment]
Hi danimo. I don't agree with or understand what you said. I'm thinking or something like a (browsable) TOC for the files and directories stored. Like in some IMAP-Clients: the initial sync only fetches the foldernames and headers of the mails. Then you can browse your folders and look at your emails. Only once you click on an email it get's downloaded.
I think this is what ferdiga (the original author of this issue) meant. What lionax meant may be a technically different and much more complicated feature.

**Comment by dragotin** [comment]
@ferdiga #3148 might help with your initial issue.

**Comment by arjamizo** [comment]
**The one reasonable solution: mounting folder as a augmented-fuse-filesystem based on `fs-cache` with owncloud's plugin. **

So looks like BitTorrent Sync has to feature in Pro version. It creates placefolder files for this. 
I agree, this requires either low file-system level integration or some kind of a proxy, which would cache the files. 

references: 
Google query `"Conduit" OR "DirSync Pro" OR "FreeFileSync" OR "Folder" OR "luckyBackup" OR "OneSync" OR "ownCloud" on-demand files`
[1] https://www.dropboxforum.com/hc/en-us/community/posts/201562035-Using-a-1TB-Dropbox-with-a-256GB-SSD-Files-on-demand
[2] https://www.getsync.com/features "Save space with placeholders"
[3] https://forum.syncthing.net/t/sync-all-sync-on-demand/5344
[4] https://documentation.commvault.com/commvault/v10/article?p=features/migration/on_demand_file_list.htm, btw. fire few times `$('.dynatree-has-children').not('.dynatree-expanded').children('span').click()` in order to find all the nodes... 
[5] https://forum.syncthing.net/t/sync-all-sync-on-demand/5344/2 (in fact useless)
[6] Reddit short discussion https://www.reddit.com/r/software/comments/2x9fub/filefolder_sync_on_demand/ **the point**: *Microsoft OneDrive had this in Windows 8.1, but it _got removed_ in Windows 10*. _Hehehehehehehehehehehhe..., funny MS_
[7] Filr 2, commercial? https://www.novell.com/communities/coolsolutions/filr-2-introduces-files-demand/ 

Google query: `"rsync" OR "Seafile community edition" OR "SparkleShare" OR "SymmetricDS" OR "Synkron" OR "krusader" OR "Unison" on-demand`
[8] http://unix.stackexchange.com/a/40109/63341 FS-cache, fuse extension used for caching files

**Comment by guruz** [comment]
-> @arjamizo https://github.com/owncloud/client/issues/4341

**Comment by RealRancor** [comment]
@arjamizo Something for you: https://github.com/owncloud/client/issues/4341

**Comment by guruz** [comment]
@RealRancor nice timing ;)
